include/deal.II/grid/grid_tools.h

```diff
@@ -1560,6 +1563,24 @@ namespace GridTools
                            const SparsityTools::Partitioner partitioner = SparsityTools::Partitioner::metis
                           );
 
+  /**
+   * This function performs the same operation as the one above, except that
+   * it takes into consideration a specific set of @p cell_weights, which allow the
+   * partitioner to balance the graph while taking into consideration the
+   * computational effort expended on each cell.
+   *
+   * @note If the @p cell_weights vector is empty, then no weighting is taken
+   * into consideration. If not then the size of this vector must equal to the
+   * number of active cells in the triangulation.
+   */
+  template <int dim, int spacedim>
+  void
+  partition_triangulation (const unsigned int                n_partitions,
+                           const std::vector<unsigned int>  &cell_weights,
```

Comment:
one option would e to move this to be the last and do `const std::vector<unsigned int>  &cell_weights = std::vector<unsigned int>()`  and then combine with the previous one. We would need to agree on who overrides who (signal vs input).

Reply:
I don't think that doing this will be any more beneficial than adding another function. Having the extra default parameter means that the user must explicitly choose the partitioner, and like you've pointed out we have to make the choice as to whether this signal overrides the input weight vector. With the current functionality this is clear. I'm going to leave this how it is.

Reply:
fair enough.